src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,504 @@
+        return delassus_matrix
+
+    @staticmethod
+    def _compute_ineq_constraint_matrix(
+        inactive_collidable_points: jtp.Vector, mu: jtp.Float
+    ) -> jtp.Matrix:
+        def compute_G_single_point(mu: float, c: float) -> jtp.Matrix:
+            """
+            Compute the inequality constraint matrix for a single collidable point
+            Rows 0-3: enforce the friction pyramid constraint,
+            Row 4: last one is for the non negativity of the vertical force
+            Row 5: contact complementarity condition
+            """
+            G_single_point = jnp.array(
+                [
+                    [1, 0, -mu],
+                    [0, 1, -mu],
+                    [-1, 0, -mu],
+                    [0, -1, -mu],
+                    [0, 0, -1],
+                    [0, 0, c],
+                ]
+            )
+            return G_single_point
+
+        G = jax.vmap(compute_G_single_point, in_axes=(None, 0))(
+            mu, inactive_collidable_points
+        )
+        G = block_diag(*G)
+        return G
+
+    @staticmethod
+    def _compute_ineq_bounds(n_collidable_points: jtp.Float) -> jtp.Vector:
+        n_constraints = 6 * n_collidable_points
+        return jnp.zeros(shape=(n_constraints,))
+
+    @staticmethod
+    def _compute_mixed_nu_dot_free(
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+        references: js.references.JaxSimModelReferences | None = None,
+    ) -> jtp.Array:
+        references = (
+            references
+            if references is not None
+            else js.references.JaxSimModelReferences.zero(
+                model=model, data=data, velocity_representation=VelRepr.Mixed
+            )
+        )
+
+        with (
+            data.switch_velocity_representation(VelRepr.Mixed),
+            references.switch_velocity_representation(VelRepr.Mixed),
+        ):
+            BW_v_WB = data.base_velocity()
+            W_o_dot_B = BW_v_WB[0:3]
+            W_omega_WB = BW_v_WB[3:6]
+            W_v̇_WB, s̈ = js.ode.system_acceleration(
+                model=model,
+                data=data,
+                joint_forces=references.joint_force_references(model=model),
+                link_forces=references.link_forces(model=model, data=data),
+            )
+
+        # Convert the inertial-fixed base acceleration to a body-fixed base acceleration.
+        W_H_B = data.base_transform()
+        W_H_BW = W_H_B.at[0:3, 0:3].set(jnp.eye(3))
+        BW_X_W = math.Adjoint.from_transform(W_H_BW, inverse=True)
+        term1 = BW_X_W @ W_v̇_WB
+        term2 = jnp.zeros(6).at[0:3].set(jnp.cross(W_o_dot_B, W_omega_WB))
+        BW_v̇_WB = term1 - term2
+
+        BW_ν̇ = jnp.hstack([BW_v̇_WB, s̈])
+
+        return BW_ν̇
+
+    @staticmethod
+    def _linear_acceleration_of_collidable_points(
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+        BW_ν̇: jax.Array,
```

Comment:
I know this is a private method, but it's not a good practice exposing unicode variables. Instead, it's ok using them in the body as it might align better the notation with the theory.

Reply:
Done in e1e0c6c9eea88e9d13b395a0a1e79bc73f620c0e